__tests__/example.test.ts

```diff
@@ -3,29 +3,170 @@ import * as testlib from '../src/validate';
+  test('Disconnect Response using actions an intent handler', async () =>{
+    const requestDisconnect = testreq.generateDisconnectRequest();
+    const responseDisconnect = fakeapp.onDisconnect(requestDisconnect);
+    const testlibValid = testlib.validate(requestDisconnect, responseDisconnect);
+||||||| merged common ancestors
     const reqSync = testreq.generateSyncRequest();
     const res = require('./example.sync.response.json');
     const testlibValid = testlib.validate(reqSync, res);
+=======
+    const responseSync = require('./example.sync.response.json');
+    const testlibValid = testlib.validate(requestSync, responseSync);
+    expect(testlibValid).toBe(undefined);
+  });
+});
+
+const devices = [{
+  'id': '123',
+  'customData': {
+    'fooValue': 74,
+    'barValue': true,
+    'bazValue': 'foo',
+  },
+}, {
+  'id': '456',
+  'customData': {
+    'fooValue': 12,
+    'barValue': false,
+    'bazValue': 'bar',
+  },
+}];
+
+const requestQuery = testreq.generateQueryRequest(devices);
+
+describe.only('Query response testing suite', () => {
+  // Test passes
+  test('Query Response passes using actions an intent handler', async () =>{
+    const responseQuery = fakeapp.onQuery(requestQuery);
+    const testlibValid = testlib.validate(requestQuery, responseQuery, responseSync);
+    expect(testlibValid).toBe(undefined);
+  });
+
+  test('Query Response fails using actions a given query response', async () =>{
+    const responseQuery = require('./example.query.response.fail.json');
+    const testlibValid = testlib.validate(requestQuery, responseQuery, responseSync);
+    expect(testlibValid).not.toBe(undefined);
+  });
+});
+
+const execution = [{
+  'command': 'action.devices.commands.OnOff',
+  'params': {
+    'on': true,
+  },
+}];
+const requestExecute = testreq.generateExecuteRequest(devices, execution);
+
+describe.only('Execute response testing suite', () => {
+  // Test should pass but there is an error here right now.
+  test('Execute Response using actions an intent handler', async () =>{
+    const responseExecute = fakeapp.onExecute(requestExecute);
+    const testlibValid = testlib.validate(requestExecute, responseExecute);
+    expect(testlibValid).toBe(undefined);
+  });
+
+  test('Execute Response fails using actions a given execute response', async () =>{
+    const responseExecute = require('./example.execute.response.fail.json');
+    const testlibValid = testlib.validate(requestExecute, responseExecute);
+    expect(testlibValid).not.toBe(undefined);
+  });
+});
+
+describe.only('Disconnect response testing suite', () => {
+  // Test should pass but there is an error here right now.
+  test('Disconnect Response using actions an intent handler', async () =>{
+    const requestDisconnect = testreq.generateDisconnectRequest();
+    const responseDisconnect = fakeapp.onDisconnect(requestDisconnect);
+    const testlibValid = testlib.validate(requestDisconnect, responseDisconnect);
+>>>>>>> Stashed changes
```

Comment:
Check this code. Make sure it builds. There are merge conflict text artifacts embedded in the file.